cmd/storage-rest-client.go

```diff
@@ -306,20 +306,20 @@ func (client *storageRESTClient) DiskInfo(ctx context.Context, opts DiskInfoOpti
 		// transport is already down.
 		return info, errDiskNotFound
 	}
-	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
 
 	opts.DiskID = client.diskID
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
 
 	infop, err := storageDiskInfoRPC.Call(ctx, client.gridConn, &opts)
```

Comment:
aren't we still doing the frequent network call here ?

Reply:
We are doing this once in a second.

Reply:
Main thing is that it is a very specific cache and any upstream changes will get us back to requesting it every time.

For example `PutObject` calls `getDiskInfos`, which returns uncached values:

https://github.com/minio/minio/blob/607cafadbc358b45163de95147c72643022ba302/cmd/erasure-server-pool.go#L1007

So every PutObject has `set_size` roundtrips before it will upload. Seems quite wasteful - even if the RPC calls are very fast :)